### ISSUE TITLE
LDAP and Email settings are not available in the chart

### ISSUE THREAD

**Issue by pierredadt** [goharbor/harbor-helm#70]
From an operator and system administrator point of view, being able to deploy a product from scratch and 100% automated is a must have.
Choosing to remove the "users" settings from the deployment chart forces an otherwise 100% automated flow to be finished manually.
This choice hence violates the kubernetes paradigm aiming at making applications deployments fully automated and reproductible.

**Comment by ywk253100** [comment]
The "user settings" should be set by API.

**Comment by neith00** [comment]
I can understand that user settings should be set by API. But in this case I can't see a smooth deprecation process. We are also using configuration through env vars for LDAP and MAIL. As far from today I wasn't warn of a deprecation nor informed that there is an on going process to define user settings through api. (By the way I don't find anything about the configuration of the ldap server and certs in the swagger) It would have been smoother to plan the deprecation to let operators the time to update their deployments by offering both ways of configuration during a defined period.
Would you give us reference where to find email server and ldap server configuration in the API?

**Comment by ywk253100** [comment]
Some refactoring works are in progress as described in https://github.com/goharbor/community/pull/12. After that, the `adminserver` will be removed and the environment variables that set user settings will not work anymore.

**Comment by ywk253100** [comment]
Closing this one, feel free to reopen it if you have any further question.

**Comment by darend** [comment]
@ywk253100 Will it still be possible to configure LDAP through the `harbor.cfg` file?

If so, another approach that allows for an automated chart installation is to separate the LDAP config from `harbor.cfg` into its own file. Users of the chart could then inject in the ldap config file through a volume mount.

Example: https://github.com/helm/charts/blob/798226aac49d8f28bca9b6afb03a89df8ce1d467/stable/datadog/values.yaml#L186-L198

**Comment by ywk253100** [comment]
The settings will be removed from `harbor.cfg`

**Comment by jimangel** [comment]
> the environment variables that set user settings will not work anymore.

@ywk253100 in [#71](https://github.com/goharbor/harbor-helm/pull/71#issuecomment-436899969) you said the reasons why email and ldap can't be configured via helm is in this PR (#71). However, the only relevant information I could find is the community proposal (not merged, blocked by @steven-zou) that states why they want to refactor the admin server. The article doesn't explain why env vars will not work anymore.

I think it makes sense to refactor the admin server (segment user and system configs) but I don't understand why the core api cannot be configured to take environment variables if present (and api interactions otherwise). This would allow 100% automation when leveraging helm to deploy harbor.

If open to my proposal, I will see if I can assist in creating that PR. Thanks!

/cc @stonezdj

**Comment by louyiping** [comment]
@ywk253100 I find a strange question.When I depoly harbor in Kubernetes successfully.I want to change the auth_mode into "ldap_auth" in UI configuration. And it's occur error and the log said "Can't load system configuration, error: system auth_mode isn't ldap_auth, please check configuration".
But if I use restful API to change it , it's ok .
Why?

**Comment by ywk253100** [comment]
@louyiping Can you open another issue to describe your problem? I cannot reproduce your issue, can you provide more information?

**Comment by kav** [comment]
Not really sure the original issue was ever addressed. As mentioned configuring the LDAP host information post install feels like it violates the k8s paradigm. I'm not sure about the approach in #71 but are you open to a PR that adds an initContainer or something to wrap the required api calls?

**Comment by R-Studio** [comment]
Any news on this?

**Comment by stonezdj** [comment]
AUTH mod and LDAP setting could be configured by Harbor configure API after installation. not exposed to values.yaml in helm chart.
